### DISCUSSION TITLE
Themes and other tweaks. Oh and hi!

### DISCUSSION THREAD

**Discussion by PinkieIcePie** [Leantime/leantime#587] [General]
Hi everyone, 
First time posting but wanted to sort of say hello :) 
I've been using Leantime for a couple months now and have been adding little bits as I go along... 
Well, I have noticed a few things on here which I have 'fixed' or added in my version and I want to let you know where I am at with some of them.

To begin with, I'm a massive fan of a dark mode on websites and apps, so I decided to initially change the CSS of leantime to match a darker theme, that was until I invited a friend of mine to help with some personal projects (which I manage with LT). He wanted an orange colour scheme, so on that note I decided to set out on a quest to build a theme system. So far the changes I have made involve adding a new column to the users table and adding a theme selection dropdown to the profile edit page. The themes are currently a hardcoded list in the dropdown of the edit profile page, however, adding another theme to it isn't difficult. In the future, I'm hoping to add a new area to the site admin section that allows the administrators to upload new themes. 

I also noticed something that felt like it was missing from the dashboard: the project description. When I first started setting up projects and users for my stuff, my friend pointed out that he couldn't see the 'project settings' menu option due his user priviliges. This meant that the information about the project was only visible to thos who could change the project settings. By copying and slightly modifying an existing function (that gets the project name) I have been able to pull the description out and render it read-only and stylised on the dashboard, I am currently working on showing some more general info about the current project on the dashboard so that it acts as an at-a-glance start page.

One thing I changed on my version that a couple of users here have mentioned is changing the Tagline on the login page to something different. I will sort out a screenshot when I'm on my PC next, currently on mobile. 

I have a long list of ideas for this, which I will slowly post about as they are completed. :) 

As I mentioned above, I will upload some screenshots of the progress as soon as possible, I just wanted to use this as an opportunity to say hi and introduce my leantime ideas :)

Hope you all enjoy what I have planned for this, I know I'm enjoying getting back into the code scene.

-Pinkie 

P.s. this is also my first time contributing to github to bare with me, I know I will make a mistake at some point...

**Comment by danshearer** [comment]
> this is also my first time contributing to github to bare with me, I know I will make a mistake at some point...

The GitHub fork/pull request model is [described quite well](https://docs.github.com/en/get-started/quickstart/fork-a-repo). I suggest you fork the repo and make your changes visible, then submit pull requests after discussion here.

Your work sounds like part of the minimum essential requirements for accessibility, so well done. Many people (almost all people, as they get older) need a high-contrast theme to see it at all. I have fellow team members for whom that is essential - and one of the biggest companies who refuse to implement basic accessibility requirements is GitHub!

--
Dan Shearer
dan@shearer.org

**Reply by PinkieIcePie** [reply]
Oooh ok, thank you for that, I'll give it a read when I get a break from the day-job 😁
